core/app/models/spree/shipment.rb

```diff
@@ -343,8 +343,8 @@ def transfer_to_location(variant, quantity, stock_location)
       transaction do
         new_shipment = order.shipments.create!(stock_location: stock_location)
 
-        order.contents.remove(variant, quantity, shipment: self)
-        order.contents.add(variant, quantity, shipment: new_shipment)
+        Spree::RemoveItemFromCart.new.call(order: order, variant: variant, quantity: quantity, options: { shipment: self })
```

Comment:
Are we going to use adjectives for naming?
I'd go with something like `Spree::Cart::Item::Remove` or `Spree::Cart::RemoveItem` for that matter, it'd be cool to group them by context / business logic domain
What do you think?